src/SFA.DAS.EmployerIncentives.Domain/Factories/IncentiveApplicationFactory.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using SFA.DAS.Common.Domain.Types;
+using SFA.DAS.EmployerIncentives.Domain.IncentiveApplications;
+using SFA.DAS.EmployerIncentives.Domain.IncentiveApplications.Models;
+
+namespace SFA.DAS.EmployerIncentives.Domain.Factories
+{
+    public class IncentiveApplicationFactory : IIncentiveApplicationFactory
+    {
+        public IncentiveApplication CreateNew(Guid id, long accountId, long accountLegalEntityId)
+        {
+            return IncentiveApplication.New(id, accountId, accountLegalEntityId);
+        }
+
+        public Apprenticeship CreateNewApprenticeship(Guid id, int apprenticeshipId, string firstName, string lastName, DateTime dateOfBirth, long uln, DateTime plannedStartDate, ApprenticeshipEmployerType apprenticeshipEmployerTypeOnApproval)
+        {
+            var model = new ApprenticeshipModel
```

Comment:
I don't think we should be creating entities this way.  The Create(model) is for hydrating from a data store.  Otherwise any domain rules could be bypassed.  So this should probably just call Create(id, FirstName, LastName, othermandatoryfields etc...)  Then the application layer (the use case) can call other methods like apprenticeship.SetPlannedStartDate(date) etc.... Then That method could check that he date is not in the past etc...  Create should just create on object the minimum required fields.

Reply:
I agree but in our case they're all minimum required fields, without any of them the object is invalid in our context.

Reply:
So should all be passed in when creating the object.